omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,222 @@
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException("Cannot execute action " + action.getAction()
+				        + " on empty set of task definitions.");
+			}
+		}
+		if (action.isAllTasks() == null || !action.isAllTasks()) {
+			for (TaskDefinition taskDef : taskDefinitions) {
+				if (taskDef == null) {
+					throw new IllegalRequestException(
+					        "One or more of the task definitions you specified are not registered on this server");
+				}
+			}
+		}
+		
+		switch (action.getAction()) {
+			case SCHEDULETASK:
+				scheduleTask(taskDefinitions);
+				break;
+			case RESCHEDULETASK:
+				reScheduleTask(taskDefinitions);
+				break;
+			case DELETE:
+				deleteTask(taskDefinitions);
+				break;
+			case SHUTDOWNTASK:
+				shutDownTask(taskDefinitions);
+				break;
+			case RESCHEDULEALLTASK:
+				reScheduleAllTask();
+				break;
+			case ONSTARTUP:
+				onStartUpTask();
+				break;
+			case ONSHUTDOWN:
+				onShutDownUpTask();
+				break;
+		}
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private void scheduleTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.scheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while scheduling task", e);
+			}
+		}
+	}
+	
+	private void shutDownTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.shutDownTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while shutdowning task", e);
+			}
+		}
+	}
+	
+	// Stop and start a scheduled task.
+	private void reScheduleTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.reScheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while rescheduling task", e);
+			}
+		}
+	}
+	
+	// Stop and start a scheduled all task.
+	private void reScheduleAllTask() {
+		try {
+			taskFactoryWrapper.reScheduleAllTasks();
+		}
+		catch (SchedulerException e) {
+			throw new APIException("Errors occurred while rescheduling all tasks", e);
+		}
+		
+	}
+	
+	// Start all tasks that are scheduled to run on startup.
+	private void onStartUpTask() {
+		taskFactoryWrapper.onStartUpTask();
+	}
+	
+	// Stop all tasks and clean up the scheduler instance.
+	private void onShutDownUpTask() {
+		taskFactoryWrapper.onShutDownUpTask();
+	}
+	
+	private void deleteTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskFactoryWrapper.unloadTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while shutdowning all tasks", e);
```

Comment:
Is this exception message correct?

Reply:
Ow sorry, Missed :-( 
Changed to "Errors occurred while deleting task".